src/content/docs/en/guides/incremental-static-regeneration.mdx

```diff
@@ -0,0 +1,276 @@
+        }
+    }
+    
+    const response = await next()
+    
+    cache.set(request.url, {
+        expires: Date.now() + expiresIn * 1000,
+        response: response.clone()
+    })
+    
+    return response
+})
+```
+:::tip
+It is a good idea to skip caching when the response code is 500 (Internal Server Error).
+:::
+
+#### Advanced: Per-page revalidation time
+You may want some pages to be always rendered fresh, while others to be kept saved for hours. Astro makes this possible with middleware locals.
+
+You can create a function that changes the `expiresIn` variable, and add it to locals. Now each astro page can decide how long it should be cached for!
+```ts title="src/middleware.ts" ins={9}
+import { defineMiddleware } from "astro:middleware"
+
+const cache = new Map
+
+export const onRequest = defineMiddleware(async ({ request, locals }, next) => {
+    
+    let expiresIn = 60
+    
+    locals.revalidate = seconds => { expiresIn = seconds }
+    
+    const cached = cache.get(request.url)
+    
+    if (cached) {
+        if (cached.expires > Date.now()) {
+            return cached.response.clone()
+        }
+        else {
+            cache.delete(request.url)
+        }
+    }
+    
+    const response = await next()
+    
+    cache.set(request.url, {
+        expires: Date.now() + expiresIn * 1000,
+        response: response.clone()
+    })
+    
+    return response
+})
+```
+```ts title="src/env.d.ts"
+/// <reference types="astro/client" />
+
+namespace App {
+    interface Locals {
+        revalidate(seconds: number): void
+    }
+}
+```
+```astro title="src/pages/index.astro"
+---
+Astro.locals.revalidate(3600)
+---
+<h1>This page will saved and reused for 1 hour before being generated again.</h1>
+```
+```ts title="src/pages/endpoint.ts"
+export async function GET({ locals }) {
+    locals.revalidate(0)
+    return new Response("This response will be generated every time this endpoint is called.")
+}
+```
+
+### Self-hosted Deno
+The middleware approach used for Node works on Deno as well. Deno also implements the Web Cache API which has the added bonus of persisting the cache across server restarts. It is important to note that `Cache-Control` headers are not automatically processed - you would have to manually delete responses when they have expired. The following middleware uses the Web Cache API to cache responses. A custom header (`X-Expires`) is used to keep track of staleness:
+
+```ts title="src/middleware.ts"
+import { defineMiddleware } from "astro:middleware"
+
+const cache = await caches.open('astro')
+
+export const onRequest = async ({ request }, next) => {
+    const cachedResponse = await cache.match(request)
+    
+    if (cachedResponse) {
+    
+        const expires = Number(cachedResponse.headers.get('X-Expires'))
+        
+        if (expires > Date.now()) {
+            return cachedResponse
+        }
+        
+        else {
+            await cache.delete(request)
+        }
+    }

```

Comment:
```suggestion
    if (cachedResponse) {
        const expires = Number(cachedResponse.headers.get('X-Expires'))
        if (expires > Date.now()) {
            return cachedResponse
        } else {
            await cache.delete(request)
        }
    }
```

Reply:
You're right, it was for readability - in my editor. I will change that to align with the convention, if we have one.